File: Problems/Algorithms/169. Majority Element/MajorityElement.kt
class Solution {
    fun majorityElement(nums: IntArray): Int {
        var majority = -1
        
        var count = 0
        for (num in nums) {
            if (count == 0) {
                majority = num
                count = 1
            } else if (majority == num) {
                count++
            } else {
                count--
            }
        }
        
        return majority
    }
}

File: Problems/Algorithms/169. Majority Element/majority-element.cpp
#include <vector>
#include <map>

using namespace std;

class Solution {
public:
    int majorityElement(vector<int>& nums) {
        map<int, int> counts;
        int n = nums.size() / 2;
        for (auto num : nums) {
            counts[num]++;
        }

        int ans = 0;
        for (auto num : counts) {
            if (num.second > n) {
                ans = num.first;
            }
        }

        return ans;
    }
};
